zarr/n5.py

```diff
@@ -304,13 +304,18 @@ def invert_chunk_coords(key):
 def group_metadata_to_n5(group_metadata):
     '''Convert group metadata from zarr to N5 format.'''
     del group_metadata['zarr_format']
+    # TODO: This should only exist at the top-level
     group_metadata['n5'] = '2.0.0'
     return group_metadata
 
 
 def group_metadata_to_zarr(group_metadata):
     '''Convert group metadata from N5 to zarr format.'''
-    del group_metadata['n5']
+    try:
+        del group_metadata['n5']
+    except KeyError:
+        # This only exists at the top level
+        pass
```

Comment:
Would suggest using `pop`. This just removes the key and returns the corresponding value. As it is a Python `dict`, this operation is pretty cheap.

```suggestion
    group_metadata.pop('n5')
```

Reply:
`.pop()` still throws a key error though, so I've re-added that handling.

Reply:
```
>>> def a():
...     dict().pop(1)
...
>>> def b():
...     del dict()[1]
...
>>> import dis
>>> dis.dis(a)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_METHOD              1 (pop)
              6 LOAD_CONST               1 (1)
              8 CALL_METHOD              1
             10 POP_TOP
             12 LOAD_CONST               0 (None)
             14 RETURN_VALUE
>>> dis.dis(b)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_CONST               1 (1)
              6 DELETE_SUBSCR
              8 LOAD_CONST               0 (None)
             10 RETURN_VALUE

¯\_(ツ)_/¯
```

Reply:
`.pop('n5', None)` if you want a default value when the key does not exists.

Reply:
Yeah sorry forgot about `None`. Submitted PR ( https://github.com/zarr-developers/zarr-python/pull/655 ) which does this.

Reply:
For completeness:
```
>>> import dis
>>> def c():
...     dict().pop(1, None)
...
>>> dis.dis(c)
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 LOAD_METHOD              1 (pop)
              6 LOAD_CONST               1 (1)
              8 LOAD_CONST               0 (None)
             10 CALL_METHOD              2
             12 POP_TOP
             14 LOAD_CONST               0 (None)
             16 RETURN_VALUE
```